src/background/retrieval-market/Provider.js

```diff
@@ -115,9 +126,21 @@ class Provider {
     });
   };
 
-  async handleNewDeal({ dealId, cid, params }, sink) {
-    ports.postLog("DEBUG: Provider.handleNewDeal()")
-    ports.postLog(`DEBUG: handling new deal ${dealId}`);
+  setOrVerifyPaymentChannel(message) {
```

Comment:
why not to split in two functions to be more clear?

Reply:
Yeah, this is probably a good point.  I logged them as TODO's now so I can merge this code without changes that may break it (it's rather fragile :-)), but I would like to fix this eventually.  Also marked the code duplication between `dealStatuses.paymentSent` and `dealStatuses.lastPaymentSent`.